build_runner_core/lib/src/environment/create_merged_dir.dart

```diff
@@ -39,17 +39,34 @@ Future<bool> createMergedOutputDirectories(
         'requested.');
     return false;
   }
+  var conflictingOutputs = _conflicts(outputLocations);
+  if (conflictingOutputs.isNotEmpty) {
+    _logger.severe('Unable to create merged directory. '
+        'Conflicting outputs for $conflictingOutputs');
+    return false;
+  }
 
-  for (var output in outputMap.keys) {
-    if (!await _createMergedOutputDir(output, outputMap[output], packageGraph,
-        environment, reader, finalizedAssetsView, outputSymlinksOnly)) {
-      _logger.severe('Unable to create merged directory for $output.');
-      return false;
+  for (var input in outputLocations.keys) {
+    for (var output in outputLocations[input]) {
+      if (!await _createMergedOutputDir(output, input, packageGraph,
+          environment, reader, finalizedAssetsView, outputSymlinksOnly)) {
+        _logger.severe('Unable to create merged directory for $output.');
+        return false;
+      }
     }
   }
   return true;
 }
 
+Set<String> _conflicts(Map<String, Set<String>> outputLocations) {
+  final seen = <String>{};
+  final conflicts = <String>{};
+  for (var location in outputLocations.values.expand((l) => l)) {
+    (seen.contains(location) ? conflicts : seen).add(location);
```

Comment:
nit: This reads better imo: `if (!seen.add(location)) conflicts.add(location);`

Reply:
ah, I completely forgot that `.add` returns a bool. Yeah this is way better.